### PR TITLE
fix: namespace issue in tfy-notebook-controller-chart

### PR DESCRIPTION
**Pull request by akashg3627**

### CODE REVIEW

**Review by nikp1172 — APPROVED** [review]

**Review by dunefro — COMMENTED** [review]
If you are making this change, let's make all these separate files and derive the values from `values.yaml` file

**Review by dunefro — APPROVED** [review]
This looks good on the whole level just the fact some file are named directly `role.yaml`, If possible please make it `notebook-controller-role.yaml`,
Similarly for `leader-election-rolebinding.yaml` and `notebook-controller-rolebinding.yaml`

File: charts/tfy-notebook-controller/values.yaml [hunk 1]
```diff
@@ -1 +1,15 @@
-
+labels:
+  app: notebook-controller
+  kustomize.component: notebook-controller
+namespace: '{{ .Release.Namespace }}'
```
Comment (dunefro) [inline charts/tfy-notebook-controller/values.yaml:4]: this is not needeed

File: charts/tfy-notebook-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+              path: /readyz
+              port: 8081
+            initialDelaySeconds: 5
+            periodSeconds: 10
+      serviceAccountName: notebook-controller-service-account
+      tolerations:
+        - key: CriticalAddonsOnly
+          value: "true"
+          effect: NoSchedule
+          operator: Equal
```
Comment (dunefro) [inline charts/tfy-notebook-controller/templates/deployment.yaml:78]: Let's move these tolerations out if possible. We will pass them separately.

File: charts/tfy-notebook-controller/templates/cluster-role-binding.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  {{- end }}
+  name: notebook-controller-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: notebook-controller-role
+subjects:
+  - kind: ServiceAccount
+    name: notebook-controller-service-account
+    namespace: {{ tpl .Values.namespace . }}
```
Comment (dunefro) [inline charts/tfy-notebook-controller/templates/cluster-role-binding.yaml:16]: You can directly pass the release.namespace here

**Pull request merged**